File: mobile/src/main/java/com/example/android/uamp/QueueHelperRealm.java
package com.example.android.uamp;


import android.content.Context;
import android.os.Bundle;
import android.support.v4.media.MediaMetadataCompat;
import android.support.v4.media.session.MediaSessionCompat;

import com.example.android.uamp.model.MusicProvider;
import com.example.android.uamp.utils.QueueHelper;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class QueueHelperRealm implements QueueHelper {
    private List<MediaSessionCompat.QueueItem> playingQueue = new ArrayList<MediaSessionCompat.QueueItem>();

    @Override

    public List<MediaSessionCompat.QueueItem> getPlayingQueue(String mediaId, MusicProvider musicProvider) {
        MediaMetadataCompat metadataCompat = musicProvider.getMusic(mediaId);
        MediaSessionCompat.QueueItem item = new MediaSessionCompat.QueueItem(metadataCompat.getDescription(), playingQueue.size() + 1);
        playingQueue.add(item);
        return Collections.unmodifiableList(playingQueue);
    }

    @Override
    public List<MediaSessionCompat.QueueItem> getPlayingQueueFromSearch(String query, Bundle queryParams, MusicProvider musicProvider) {
        throw new UnsupportedOperationException();
    }

    @Override
    public int getMusicIndexOnQueue(Iterable<MediaSessionCompat.QueueItem> queue, String mediaId) {
        return 0;
    }

    @Override
    public int getMusicIndexOnQueue(Iterable<MediaSessionCompat.QueueItem> queue, long queueId) {
        return 0;
    }

    @Override
    public List<MediaSessionCompat.QueueItem> getRandomQueue(MusicProvider musicProvider) {
        return null;
    }

    @Override
    public boolean isIndexPlayable(int index, List<MediaSessionCompat.QueueItem> queue) {
        return false;
    }

    @Override
    public boolean equals(List<MediaSessionCompat.QueueItem> list1, List<MediaSessionCompat.QueueItem> list2) {
        return false;
    }

    @Override
    public boolean isQueueItemPlaying(Context context, MediaSessionCompat.QueueItem queueItem) {
        return false;
    }
}

File: mobile/src/main/java/com/example/android/uamp/QueueHelperRealm.kt
package com.example.android.uamp


import android.content.Context
import android.os.Bundle
import android.support.v4.media.session.MediaSessionCompat
import com.example.android.uamp.model.MusicProvider
import com.example.android.uamp.utils.QueueHelper
import java.util.*

internal class QueueHelperRealm : QueueHelper {
    private val playingQueue = ArrayList<MediaSessionCompat.QueueItem>()

    override fun getPlayingQueue(mediaId: String, musicProvider: MusicProvider): List<MediaSessionCompat.QueueItem> {
        val metadataCompat = musicProvider.getMusic(mediaId)
        val item = MediaSessionCompat.QueueItem(metadataCompat.description, (playingQueue.size + 1).toLong())
        playingQueue.add(item)
        return Collections.unmodifiableList<MediaSessionCompat.QueueItem>(playingQueue)
    }

    override fun getPlayingQueueFromSearch(query: String, queryParams: Bundle, musicProvider: MusicProvider): List<MediaSessionCompat.QueueItem> {
        throw UnsupportedOperationException()
    }

    override fun getMusicIndexOnQueue(queue: Iterable<MediaSessionCompat.QueueItem>, mediaId: String): Int {
        return 0
    }

    override fun getMusicIndexOnQueue(queue: Iterable<MediaSessionCompat.QueueItem>, queueId: Long): Int {
        return 0
    }

    override fun getRandomQueue(musicProvider: MusicProvider): List<MediaSessionCompat.QueueItem>? {
        return null
    }

    override fun isIndexPlayable(index: Int, queue: List<MediaSessionCompat.QueueItem>): Boolean {
        return (queue?.size > 0) && (queue?.size >= index)
    }

    override fun equals(list1: List<MediaSessionCompat.QueueItem>, list2: List<MediaSessionCompat.QueueItem>): Boolean {
        return false
    }

    override fun isQueueItemPlaying(context: Context, queueItem: MediaSessionCompat.QueueItem): Boolean {
        return false
    }
}
